doc/api/assert.md

```diff
@@ -45,12 +45,12 @@ Only [enumerable "own" properties][] are considered. The
 non-enumerable properties — for such checks, consider using
 [`assert.deepStrictEqual()`][] instead. This can lead to some
 potentially surprising results. For example, the following example does not
-throw an `AssertionError` because the properties on the [`Error`][] object are
+throw an `AssertionError` because the properties on the [`RegExp`][] object are
 not enumerable:
 
 ```js
 // WARNING: This does not throw an AssertionError!
-assert.deepEqual(Error('a'), Error('b'));
+assert.deepEqual(/a/, {});
```

Comment:
I wonder if we can either come up with a better example or if it's OK to just remove the example entirely. The current one makes it so clear that intuition would be wrong, but this one not so much. This seems more akin to `assert.deepEqual('', 0)` than the current example.